node/src/store/in_memory_store.rs

```diff
@@ -115,7 +115,7 @@ impl InMemoryStore {
             return false;
         };
 
-        height <= head_height
+        height != 0 && height <= head_height
```

Comment:
get_head_height returns an error for 0 height already

Reply:
This covers the case of already filled store. Header with height zero was then incorrectly reported as present here, since 0 < head_height.